.circleci/config.yml

```diff
@@ -518,91 +532,91 @@ defaults:
       python2: true
 
   - job_native_test_ext_gnosis: &job_native_test_ext_gnosis
-      <<: *workflow_ubuntu2204_static
+      <<: *workflow_ubuntu2004_static
```

Comment:
It's actually weird that we even include the version here. It's misleading IMO - it looks like it should make the job run on Ubuntu 20.04 but what it really does is just add a dependency on `b_ubu_static`. It's `b_ubu_static` that happens to run on 20.04 and that's completely superfluous information. It would also be pretty easy to forget to update the name when moving a job to a new base image.

I think we would be better off renaming these things. My suggestion would be:
- For the ones that filter branches:
    - `workflow_trigger_on_tags` -> `on_all_tags_and_branches`
    - `workflow_trigger_on_releases` -> `on_version_tags`
    - A new one for nightly that's currently specified directly: `on_develop`
- For the ones that specify dependencies:
    - `workflow_ubuntu2004_static` -> `requires_b_ubu_static`
    - `workflow_ubuntu2204_asan_clang` -> `requires_b_ubu_asan_clang`
    - ...
    - For ones that do not require anything add `requires_nothing` which is just  `on_all_tags_and_branches` with no extra fields.

This is out of scope of this PR of course, but you could include this change if you decide to do that base image cleanup PR I mentioned in another comment.

Reply:
Nice. I will do that in the new PR then. But `requires_nothing` is a bit weird no? Maybe we can just make a `default_workflow` or something like that.

Reply:
Yeah, no, I don't have a better suggestion, maybe just `no_dependencies` haha
But yeah, I will use your suggestion in the new PR (i.e. `requires_nothing`), and we can think on a better name later if needed.

Reply:
> maybe just `no_dependencies` haha

I thought about that one too and I'd be fine with it. I just thought that `requires_nothing` would be more inline with the other `requires_xyz` dicts we would get after the renaming.